18F/identity-idp
# Reintroduce Propshaft as replacement for Sprockets

## 🛠 Summary of changes

Replaces [Sprockets](https://github.com/rails/sprockets) with [Propshaft](https://github.com/rails/propshaft) for managing the asset pipeline.

This is a retry of #8387, which was reverted in #8412 due to issues observed in deployed environments.

The key change here is in 1474ff044d871c20faf2d329b0ee503ad3474b5f. By removing the configuration of `asset_host`, it's hoped to avoid issues which lead to the use of an [inaccurate URL prefix in the published version of the Propshaft gem](https://github.com/rails/propshaft/blob/v0.7.0/lib/propshaft/compilers/css_asset_urls.rb#L10). The `asset_host` configuration is not necessary because (a) our CDN is now configured to serve from the same host as the IdP domain, and (b) Propshaft will still generate root-relative paths (not absolute paths) without the configuration.

## 📜 Testing Plan

1. Go to http://localhost:3000/
2. Observe that images and styles appear as expected

(Further testing will be conducted in deployed environments once merged)
